### PR TITLE
Add iOS examples

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

**Comment by keith** [comment]
had to do a new CI job here because of https://github.com/bazelbuild/bazel/issues/16099

File: examples/ios/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+rust_library(
+    name = "rust_lib",
+    srcs = ["demo.rs"],
+    edition = "2018",
+    tags = ["manual"],
+    deps = [":allocator_library"],
+)
+
+# This being required is a bug in objc_library, ideally 'main_lib' could depend directly on 'rust_lib'
```
Comment (UebelAndre) [inline examples/ios/BUILD.bazel:20]: Do you know if this is tracked by an upstream issue in Bazel? Or should one be filed for `rules_rust`?

File: examples/ios/platform_mappings [hunk 2]
```diff
@@ -0,0 +1,20 @@
+flags:
```
Comment (UebelAndre) [inline examples/ios/platform_mappings:1]: Why are platform mappings needed? Could transitions not be used?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thanks! Looks good to me but just had some questions

File: examples/ios/platform_mappings (hunk 2 above)
Comment (keith) [reply examples/ios/platform_mappings:1]: the apple rules don't currently support platforms, so the transitions from those rules don't enforce a platform on their dependencies yet

File: examples/ios/BUILD.bazel (hunk 1 above)
Comment (keith) [reply examples/ios/BUILD.bazel:20]: https://github.com/bazelbuild/bazel/pull/15462

Comment (UebelAndre) [reply examples/ios/BUILD.bazel:20]: Can we add that the code here? Something like the following maybe?:
```suggestion
# TODO(bazelbuild/bazel#15462): This being required is a bug in objc_library, ideally 'main_lib' could depend directly on 'rust_lib'
```

File: examples/ios/platform_mappings (hunk 2 above)
Comment (UebelAndre) [reply examples/ios/platform_mappings:1]: Is there a tracking ticket for this somewhere?

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: examples/ios/BUILD.bazel (hunk 1 above)
Comment (keith) [reply examples/ios/BUILD.bazel]: Adding a TODO to remove, I verified it works without this shim with the latest rolling release

File: examples/ios/platform_mappings (hunk 2 above)
Comment (UebelAndre) [reply examples/ios/platform_mappings:2]: Was a ticket made but not linked?

Comment (keith) [reply examples/ios/platform_mappings:2]: it's on the first line in this file?

Comment (UebelAndre) [reply examples/ios/platform_mappings:2]: Ah! https://github.com/bazelbuild/rules_apple/issues/1658

Thank you! Sorry for the confusion 😅

**Review by UebelAndre — APPROVED** [review]

File: examples/ios/platform_mappings (hunk 2 above)
Comment (keith) [reply examples/ios/platform_mappings:2]: npnp <3

**Pull request merged**